snuba_sdk/mql/mql.py

```diff
@@ -53,16 +52,17 @@
 variable = "$" ~r"[a-zA-Z0-9_.]+"
 nested_expression = open_paren _ expression _ close_paren
 
-function = (aggregate / curried_aggregate / arbitrary_function) (group_by)?
+function = (curried_aggregate / aggregate / arbitrary_function) (group_by)?
 aggregate = aggregate_name (open_paren _ filter _ close_paren)
-curried_aggregate = curried_aggregate_name (open_paren _ aggregate_list? _ close_paren) (open_paren _ filter _ close_paren)
-arbitrary_function = arbitrary_function_name (open_paren ( _ filter _ ) (_ comma _ param_expression)* close_paren)
+aggregate_name = "avg" / "count" / "max" / "min" / "sum" / "last" / "uniq"
+arbitrary_function = arbitrary_function_name (open_paren ( _ expression _ ) (_ comma _ expression)* close_paren)
+arbitrary_function_name = ~r"[a-zA-Z0-9_]+"
+curried_aggregate = curried_aggregate_name (open_paren _ aggregate_list? _ close_paren) (open_paren _ expression (_ comma _ expression)* _ close_paren)
```

Comment:
I don't think this is right. A curried_aggregate should only have one parameter, which is a string. So `quantiles(0.5)(transaction.duration)` is a curried aggregate, but `quantiles(0.5)(transaction.duration, 10)` is not.

Reply:
I suspect you will need a `curried_arbitrary_function` to support expressions like `topK(10)(sum(transaction.duration))`

Reply:
That makes sense. I can add `curried_arbitrary_function`.